src/EditorFeatures/Core/Shared/Utilities/ThreadingContext.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Utilities;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
+{
+    [Export(typeof(IThreadingContext))]
+    [Shared]
+    internal sealed partial class ThreadingContext : IThreadingContext
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ThreadingContext([Import(AllowDefault = true)] JoinableTaskContext joinableTaskContext)
```

Comment:
If the editor we're sitting atop now requires this, why default this? Who are we hurting?

Reply:
If we are certain that prior to this change the use of Roslyn's Editor Features layer requires the MEF catalog to contain a `JoinableTaskContext`, then I can make this required. This was a carefully-constructed type intended to allow the use of `JoinableTaskContext` without creating a new hidden dependency on the MEF catalog.

Reply:
➡️ Added to the list of follow-up items.